### PR TITLE
chore: Added python 3.12 image.

### PR DESCRIPTION
**Pull request by cb-sl**
Added Python 3.12 image to ensure coverage before publishing policy.

### CODE REVIEW

File: linux [hunk 1]
```diff
@@ -40,8 +40,9 @@ python:3.8 python-3.8
...
 python:3.10 python-3.10
 python:3.11 python-3.11
+python:3.12 python-3.12
 ruby
 ruby:3.3 ruby-3.3
 swift
 ubuntu linux
-sbtscala/scala-sbt:eclipse-temurin-jammy-22_36_1.10.0_3.4.2 sbt1.10.0-scala3.4.2
\ No newline at end of file
+sbtscala/scala-sbt:eclipse-temurin-jammy-22_36_1.10.0_3.4.2 sbt1.10.0-scala3.4.2
```
Comment (PeterSchafer) [inline linux:48]: Issue: I don't think this was intentional to be re added

**Review by PeterSchafer — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -12,6 +12,10 @@ on:
     - "*"
     - "!README.md"
     - "!build.rb"
+  schedule:
```
Comment (PeterSchafer) [inline .github/workflows/build.yml:15]: Issue: This was also recently removed.

File: linux (hunk 1 above)
Comment (cb-sl) [reply linux:48]: This is not my change and I was also surprised to see this. Could we remove this, please?

File: .github/workflows/build.yml (hunk 2 above)
Comment (cb-sl) [reply .github/workflows/build.yml:15]: Yup, am not sure why this happened. Happy to pair?

Comment (cb-sl) [reply .github/workflows/build.yml:15]: Again, this is an automated process - the command that I run is the following.

docker run -it --rm -v $(pwd):/app ruby:2.6-slim bash -c "cd app && ruby build.rb"

I wonder if this adds the schedule back via build.rb (am speculating only)

Comment (cb-sl) [reply .github/workflows/build.yml:15]: :D I think it is coming from this https://github.com/snyk/snyk-images/blob/e0b1c2c00b30994fe6276d01cb4a91358d5cbd5a/_templates/build.yml.erb#L15 

Maybe we need to remove it from here as well...

**Comment by cb-sl** [comment]
Note to self: Maybe edit linux file as well as build.yml.erb to ensure this side effect does not happen again for others. To check with Peter. Parked for now.

File: linux (hunk 1 above)
Comment (cb-sl) [reply linux:48]: No, wasn't

**Pull request closed without merging**